### ISSUE TITLE
 [Swift3] Bug when handling java date 

### ISSUE THREAD

**Issue by zxy198717** [swagger-api/swagger-codegen#4318]
##### Description

`Java date` cannot covert to `swift3 date` on iPhone 5C

##### Swagger-codegen version

Master branch

##### Swagger declaration file content or url

https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/swift3/Models.mustache
Line 144
```
if let sourceInt = source as? Int {
      // treat as a java date
      return Date(timeIntervalSince1970: Double(sourceInt / 1000) )
}
```
##### Suggest a Fix

```
if let sourceInt = source as? Int64 {
      // treat as a java date
      return Date(timeIntervalSince1970: Double(sourceInt / 1000) )
}
```

And suggest to add simple date format `yyyy-MM-dd HH:mm:ss`:
```
let formatters = [
            "yyyy-MM-dd",
            "yyyy-MM-dd'T'HH:mm:ssZZZZZ",
            "yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ",
            "yyyy-MM-dd'T'HH:mm:ss'Z'",
            "yyyy-MM-dd'T'HH:mm:ss.SSS",
            "yyyy-MM-dd HH:mm:ss"
        ].map { (format: String) -> DateFormatter in
            let formatter = DateFormatter()
            formatter.dateFormat = format
            return formatter
        }
```

**Comment by wing328** [comment]
@zxy198717 do you mind submitting a PR so that we can review the change?

**Comment by zxy198717** [comment]
@wing328 Done.

**Comment by wing328** [comment]
@zxy198717 ok. I'll run some tests.

(If your company is using Swagger Codegen, please consider adding your company name to [this list](https://github.com/swagger-api/swagger-codegen#companiesprojects-using-swagger-codegen))

**Comment by wing328** [comment]
Closed via #4332
